config-model-api/src/main/java/com/yahoo/config/application/api/DeploymentSpec.java

```diff
@@ -57,10 +58,12 @@
     private final Optional<AthenzDomain> athenzDomain;
     private final Optional<AthenzService> athenzService;
     private final Notifications notifications;
+    private final List<Endpoint> endpoints;
 
     public DeploymentSpec(Optional<String> globalServiceId, UpgradePolicy upgradePolicy, Optional<Integer> majorVersion,
                           List<ChangeBlocker> changeBlockers, List<Step> steps, String xmlForm,
-                          Optional<AthenzDomain> athenzDomain, Optional<AthenzService> athenzService, Notifications notifications) {
+                          Optional<AthenzDomain> athenzDomain, Optional<AthenzService> athenzService, Notifications notifications,
+                          List<Endpoint> endpoints) {
         validateTotalDelay(steps);
         this.globalServiceId = globalServiceId;
```

Comment:
Constructor should throw if `globalServiceId.isPresent() == !endpoints.isEmpty()` as both cannot be set.